tools/lib/h5diff.c

```diff
@@ -1117,11 +1094,26 @@ diff_match(hid_t file1_id, const char *grp1, trav_info_t *info1, hid_t file2_id,
     char       *obj1_fullpath = NULL;
     char       *obj2_fullpath = NULL;
     diff_args_t argdata;
-    size_t      idx1      = 0;
-    size_t      idx2      = 0;
-    diff_err_t  ret_value = opts->err_stat;
+    size_t      idx1 = 0;
+    size_t      idx2 = 0;
+#ifdef H5_HAVE_PARALLEL
+    char *workerTasks = NULL;
+    int   busyTasks   = 0;
+#endif
+    diff_err_t ret_value = opts->err_stat;
 
     H5TOOLS_START_DEBUG(" - errstat:%d", opts->err_stat);
+
+#ifdef H5_HAVE_PARALLEL
+    if (g_Parallel) {
+        if (NULL == (workerTasks = malloc((size_t)(g_nTasks - 1) * sizeof(char))))
```

Comment:
Added error checking to allocation of worker tasks array and moved it up to fail early on failed allocation.